log/filelogger/log.go

```diff
@@ -26,30 +27,30 @@ const (
 	maxFiles = 50
 )
 
-// New returns a logf wrapper that appends to local disk log
+// New returns a Writer that appends to local disk log
 // files on Windows, rotating old log files as needed to stay under
 // file count & byte limits.
-func New(fileBasePrefix, logID string, logf logger.Logf) logger.Logf {
+func New(fileBasePrefix, logID string, inner *log.Logger) io.Writer {
```

Comment:
so, I like everything about this PR (like moving the file logging up from ipnserver) except this signature change. An io.Writer is logically a byte stream and not record oriented like logf is. Generally when an io.Writer is used record-wise (or a passed in one has that constraint), it's documented, like the `log` package saying:

> Each logging operation makes a single call to the Writer's Write method.

That'd be one option, to just document it here.

But I'd prefer to stay in the `logf` type world if possible.

I see why that's tricky, though... logpolicy is what generates the LogID.

Still thinking.

Reply:
What do you think about #3582?